### PR TITLE
feat(datepicker): allow to select dates from other months

### LINKED ISSUE #3746: Datepicker - Should be able to select a date from early next month without changing months
### Bug description or feature request:
If today is 28th Feb and I can only select future dates, I want to choose 1st March. However I have to click on the move to next month button before I can do anything. Currently next month's dates are all disabled.

It would be better if I could click 1st March whilst in the month of February, and have it automatically switch to showing March. This is how most date pickers work. It would save a click.

### Plunker/StackBlitz that reproduces the issue:
You can see the issue on the official demo: https://valor-software.com/ngx-bootstrap/#/datepicker

### LINKED ISSUE #4485: Datepicker: allow selection for days from adjacent months.
### Feature request:
Datepicker doesn't allow selection of days that don't belong to the displayed month.
This behavior is hard-coded [here](https://github.com/valor-software/ngx-bootstrap/blob/b4c0bcb18732713c4732992b6987ed1d11990941/src/datepicker/themes/bs/bs-daterangepicker-container.component.ts#L64) for date range picker and [here](https://github.com/valor-software/ngx-bootstrap/blob/b4c0bcb18732713c4732992b6987ed1d11990941/src/datepicker/themes/bs/bs-datepicker-container.component.ts#L62) for single date picker.

It would be great to have an option in datepicker config to allow this.
Ideally picker would change the current month as well to the one the selected day belongs to.

Styling may still indicate that the day is from another month.

### PR DESCRIPTION
**Pull request by daniloff200**
closes #4485
closes #3746

A new property for BsConfig was added : **selectFromOtherMonth**

# PR Checklist
Before creating new PR, please take a look at checklist below to make sure that you've done everything that needs to be done before we can merge it.

 - [x] read and followed the [CONTRIBUTING.md](https://github.com/valor-software/ngx-bootstrap/blob/development/CONTRIBUTING.md) guide.
 - [x] built and tested the changes locally.
 - [x] added/updated tests.
 - [x] added/updated API documentation.
 - [x] added/updated demos.

### CODE REVIEW

File: src/datepicker/themes/bs/bs-datepicker-container.component.ts [hunk 1]
```diff
@@ -61,8 +62,14 @@ export class BsDatepickerContainerComponent extends BsDatepickerAbstractComponen
   }
 
   daySelectHandler(day: DayViewModel): void {
-    if (day.isOtherMonth || day.isDisabled) {
-      return;
+    if (this.isOtherMonthsActive) {
```
Comment (Domainv) [inline src/datepicker/themes/bs/bs-datepicker-container.component.ts]: "if" inside "if" is bad practice.
I suggest using for example:

```
const isDisabled = this.isOtherMonthsActive ? day.isDisabled : (day.isOtherMonth || day.isDisabled);

    if (isDisabled) {
      return;
    }

    this._store.dispatch(this._actions.select(day.date));
```

File: src/datepicker/themes/bs/bs-daterangepicker-container.component.ts [hunk 2]
```diff
@@ -61,8 +62,14 @@ export class BsDaterangepickerContainerComponent extends BsDatepickerAbstractCom
   }
 
   daySelectHandler(day: DayViewModel): void {
-    if (day.isOtherMonth || day.isDisabled) {
-      return;
+    if (this.isOtherMonthsActive) {
+      if (day.isDisabled) {
```
Comment (Domainv) [inline src/datepicker/themes/bs/bs-daterangepicker-container.component.ts]: same

Comment (daniloff200) [reply src/datepicker/themes/bs/bs-daterangepicker-container.component.ts]: done

**Review by Domainv — APPROVED** [review]

File: src/datepicker/bs-datepicker.config.ts [hunk 3]
```diff
@@ -21,7 +21,10 @@ export class BsDatepickerConfig
    * Default max date for all date/range pickers
    */
   maxDate?: Date;
-
+  /**
+   * Makes dates from other months active
+   */
+  isOtherMonthsDatesActive = false;
```
Comment (Domainv) [inline src/datepicker/bs-datepicker.config.ts]: Need to rename similar to other libraries

**Review by Domainv — CHANGES_REQUESTED** [review]

**Comment by ludmilanesvitiy** [comment]
New Demo for Datepicker Tested in CHrome, FF - latest, looks good.

**Review by Domainv — APPROVED** [review]

**Review by valorkin — APPROVED** [review]
LGTM

**Pull request merged**